File: playground/BOJ/boj_2231/boj_2231.js
/*
----------------------------------------------
Sub : [BOJ] 분해합
Link: https://www.acmicpc.net/problem/2231
Tag : JS, Math
Memo
----------------------------------------------
*/

const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const N = Number(input[0])

let [cnt, minN] = [N, N]

// 생성자를 반환하는 함수
const devideSum = (num) => {
  const el = num.toString();
  let elSum = num
  for (let i = 0; i < el.length; i++) {
    elSum += Number(el[i])
  }

  if (elSum === N) return num

  return 0
}

while (cnt > N / 2) {
  cnt -= 1
  let cur = devideSum(cnt)
  if (cur !== 0 && cur < minN) minN = cur
}

// 출력
if (minN === N) console.log(0)
else console.log(minN)

File: playground/BOJ/boj_2231/boj_2231.py
'''
----------------------------------------------
Sub : [BOJ] 분해합
Link: https://www.acmicpc.net/problem/2231
Tag : Python, 
Memo
----------------------------------------------
'''

N = int(input())
cnt = N
minN = N

# 생성자를 반환하는 함수
def devideSum(num):
    el = str(num)
    elSum = num

    for _ in el:
        elSum += int(_)

    if elSum == N:
        return num
    
    return 0


while cnt > N/2:
    cnt -= 1
    cur = devideSum(cnt)
    
    if cur != 0 and cur < minN :
        minN = cur

# 출력
if minN == N:
    print(0)
else :
    print(minN)


